### PR TITLE
Rename styleguide and component library app -> Component Catalog

### PR DESCRIPTION
**Pull request by tesk9**
Fixes A11-2297

Big naming changes! We're rebranding the "styleguide" as the "Component Catalog"

### Before

Screen Shot 2023-02-13 at 3 48 17 PM

### After

Screen Shot 2023-02-13 at 3 47 51 PM

cc @NoRedInk/design

### CODE REVIEW

**Comment by bendansby** [comment]
This is big!

File: .github/pull_request_template.md
Comment (krinhorn) [inline .github/pull_request_template.md:57]:
```suggestion
        - if you want to test the Component Catalog example directly, add ProgramTest-style tests under `component-catalog/tests`
```

File: Shakefile.hs
Comment (krinhorn) [inline Shakefile.hs:74]:
```suggestion
        -- so that we're not presenting outdated stuff in the Component Catalog! This
```

File: component-catalog-app/Examples/Carousel.elm
Comment (krinhorn) [inline component-catalog-app/Examples/Carousel.elm:152]:
```suggestion
        [ -- faking a mini version of the Carousel component to give Component Catalog users a sense of what the
```

File: component-catalog-app/Examples/Modal.elm
Comment (krinhorn) [inline component-catalog-app/Examples/Modal.elm:137]:
```suggestion
        [ -- faking a mini version of the Modal component to give Component Catalog users a sense of what the
```

File: component-catalog-app/Examples/Page.elm
Comment (krinhorn) [inline component-catalog-app/Examples/Page.elm:66]:
```suggestion
        [ -- faking a mini version of the Page component to give Component Catalog users a sense of what the
```

File: component-catalog-app/Examples/Tabs.elm
Comment (krinhorn) [inline component-catalog-app/Examples/Tabs.elm:60]:
```suggestion
        [ -- faking a mini version of the Tabs component to give Component Catalog users a sense of what the
```

File: src/Nri/Ui/Button/V10.elm
Comment (krinhorn) [inline src/Nri/Ui/Button/V10.elm:530]:
```suggestion
The Component Catalog example will NOT work correctly because of <https://github.com/elm/browser/issues/34>, which describes a problem where "a tags without href generate a navigation event".
```

File: src/Nri/Ui/ClickableText/V3.elm
Comment (krinhorn) [inline src/Nri/Ui/ClickableText/V3.elm:376]:
```suggestion
The Component Catalog example will NOT work correctly because of <https://github.com/elm/browser/issues/34>, which describes a problem where "a tags without href generate a navigation event".
```

File: src/Nri/Ui/TextArea/V5.elm
Comment (krinhorn) [inline src/Nri/Ui/TextArea/V5.elm:45]:
```suggestion
## The Nri Component-Catalog-specified textarea with overlapping label
```
(I don't actually entirely know what this means, so no sure if "Nri" is needed or if it should be "NRI"? Just a flag - your call!)

File: tests/Spec/Nri/Ui/Block.elm
Comment (krinhorn) [inline tests/Spec/Nri/Ui/Block.elm:335]:
```suggestion
    , test "Component Catalog example regression test: 3 labels on 1 line, with 1 overlap to resolve" <|
```

**Review by krinhorn — APPROVED** [review]
Approving assuming suggestions will be accepted! (I think it's just find-and-replace bloopers.) Thank you!!

**Comment by tesk9** [comment]
A LOT of bloopers! 😅

**Pull request merged**
